geom/alg_simplify_test.go

```diff
@@ -123,10 +123,22 @@ func TestSimplifyErrorCases(t *testing.T) {
 			))`,
 			1e-5,
 		},
+
+		// Second case for "outer ring becomes invalid after simplification".
+		{`POLYGON((0 0,0 1,0.9 1,1 1.1,1.1 1,2 1,2 0,1 1.05,0 0))`, 0.2},
+
+		// Inner ring becomes invalid after simplification.
```

Comment:
If we know the reason for the inner ring becoming invalid, do you think it would be useful to add this reason in this comment?

Reply:
Agreed, I'll add that.

Reply:
I've added to the comment in ca0e2de732356b70b26aea2b546fff34a6074fe0, but I'm not sure it makes it _completely_ clear from the comment itself. I think ultimately with these sorts of tests, a reader (or a developer debugging!) would really need to view the polygons visually to really understand what's going on. In this case, the polygon with inner ring looks like this:

<img width="430" alt="Screen Shot 2021-11-17 at 5 46 35 am" src="https://user-images.githubusercontent.com/5352655/142046626-96c6886f-bb2c-4be2-a2e1-70d8dc52ae44.png">

The simplify algorithm would remove a single control point (the most northern point in the inner ring). That causes the inner ring to overlap on itself.

Reply:
I agree that it's hard to really understand without an image, but I like that it mentions that executing the Simplify operation causes an inner ring to intersect with itself.

Although, if we remove the northern point in the inner ring, I don't understand how the inner ring would overlap on itself. I thought it would just result in a line string within the polygon since the inner ring is no longer connected?

Reply:
> Although, if we remove the northern point in the inner ring, I don't understand how the inner ring would overlap on itself. I thought it would just result in a line string within the polygon since the inner ring is no longer connected?

So when I say "remove the northern most control point", I mean that the two control points adjacent to it will join up ("skipping" the middle one). An image of the geometry in that invalid geometry is below:

<img width="512" alt="Screen Shot 2021-11-17 at 7 51 30 am" src="https://user-images.githubusercontent.com/5352655/142063715-3e9f17cc-7134-4cab-935f-5a6bf52498cc.png">

Reply:
Aaaah, I see, it's the implicit re-joining that I wasn't aware of. Thanks!